core/src/test/java/org/infinispan/persistence/sifs/SoftIndexFileStoreFileStatsTest.java

```diff
@@ -137,16 +141,8 @@ void waitForCompletion() throws InterruptedException {
    }
 
    Map.Entry<Integer, Compactor.Stats> extractCompletedStat(ConcurrentMap<Integer, Compactor.Stats> statsMap) {
-      Map.Entry<Integer, Compactor.Stats> completedStats = null;
-      for (Map.Entry<Integer, Compactor.Stats> entry : statsMap.entrySet()) {
-         if (entry.getValue().isCompleted()) {
-            if (completedStats != null) {
-               fail("More than one stat was completed: " + statsMap);
-            }
-            completedStats = entry;
-         }
-      }
-      return completedStats;
+      return statsMap.entrySet().stream().filter(e -> e.getValue().isCompleted()).findFirst()
```

Comment:
Note that this doesn't fail any longer by having more than one stat file. The calling tests instead assert the file stats match directly, so no reason for this to do that here.